plugins/xctest-plugin/src/main/java/io/qameta/allure/xctest/XcTestPlugin.java

```diff
@@ -151,6 +147,23 @@ private void parseStep(final Path directory, final Object parent,
         lastFailedStep.map(Step::getStatusTrace).ifPresent(step::setStatusTrace);
     }
 
+    private void addAttachment(final Path directory,
+                               final ResultsVisitor visitor,
+                               final Map<String, Object> props,
+                               final Step step) {
+        String uuid = props.get("UUID").toString();
+        Path pngAttachmentPath = directory.resolve("Attachments").resolve(String.format("Screenshot_%s.png", uuid));
```

Comment:
resolve attachments directory at first to remove duplicate "Attachments" string.

Also you can search for all files by `Screenshot_%s.*` glob and add them as attachments

Reply:
Searching by `Screenshot_%s.*` didn't work, so instead i just loop over a extenstions array. Works locally for me

Reply:
did you try DirectoryStream? 

```java
try (DirectoryStream<Path> paths = Files.newDirectoryStream(attachments, String.format("Screenshot_%s.*", uuid))) {
    for (Path attachmentPath : paths) {
        if (Files.exists(attachmentPath)) {
            ...
        }
    }
} catch (IOException e) {
    ...
}
```

or you can use Stream API:

```java
Stream.of("jpg", "png")
        .map(ext -> attachments.resolve(String.format("Screenshot_%s.%s", uuid, ext)))
        .filter(Files::exists)
        .map(visitor::visitAttachmentFile)
        .forEach(step.getAttachments()::add);
```

Reply:
Agree, Stream API looks really nice. Applied it and checked locally - works like a charm :) Should i squash all commits into 1 ?

Reply:
> Should i squash all commits into 1 ?

as u wish